### PR TITLE
Multiple Fallback Provider

### PR DESCRIPTION
**Pull request by abkaplan07**
This PR creates a `LocationProvider` that uses a list of location service providers for fallbacks. The fallback list is created via a `Builder` pattern, with convenience methods for adding Google Location Services and the default Android location manager.

### CODE REVIEW

File: library/src/test/java/io/nlopez/smartlocation/location/providers/MultiFallbackProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        verifyNoMoreInteractions(providerMock);
+    }
+
+    @SafeVarargs
+    private final void checkExpectedProviders(MultiFallbackProvider subject, Class<? extends
+            LocationProvider>... expectedProviders) {
+        Collection<LocationProvider> providers = subject.getProviders();
+        assertEquals(expectedProviders.length, providers.size());
+        int i = 0;
+        for (LocationProvider provider : providers) {
```
Comment (mrmans0n) [inline library/src/test/java/io/nlopez/smartlocation/location/providers/MultiFallbackProviderTest.java]: You should be using a fori loop here, as you need to have the index anyway.

File: library/build.gradle [hunk 2]
```diff
@@ -16,6 +16,13 @@ apply plugin: 'signing'
 version '3.2.2'
 group 'io.nlopez.smartlocation'
 
+ext {
+    sonatypeRepo = ""
```
Comment (mrmans0n) [inline library/build.gradle]: You should probably add a function here to check if the values have been set via cmdline or whatever, and fallback to "" if not. That way my publishing scripts for this library won´t fail, making my life easier :)

File: library/src/main/java/io/nlopez/smartlocation/location/providers/MultiFallbackProvider.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        }
+        return current.getLastLocation();
+    }
+
+    boolean addProvider(LocationProvider provider) {
+        return providers.add(provider);
+    }
+
+    Collection<LocationProvider> getProviders() {
+        return this.providers;
```
Comment (mrmans0n) [inline library/src/main/java/io/nlopez/smartlocation/location/providers/MultiFallbackProvider.java]: nit: I´d like the ´this.´ usages to be only whenever are necessary.

**Comment by mrmans0n** [comment]
This looks very nice. I´d like to get rid of the current default provider and use this instead by default. Feel free to update SmartLocation.java class to do so.

Also, care to add some brief instructions to the README.md?

Thanks a lot!

**Comment by abkaplan07** [comment]
Unclear why the build is failing...seems to work fine on my local machine.

**Pull request merged**